examples/palace-tonegen.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python3
+# Sample for tone generator
+# Copyright (C) 2020  Ngô Ngọc Đức Huy
+#
+# This file is part of palace.
+#
+# palace is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published
+# by the Free Software Foundation, either version 3 of the License,
+# or (at your option) any later version.
+#
+# palace is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with palace.  If not, see <https://www.gnu.org/licenses/>.
+
+from argparse import Action, ArgumentParser
+from array import array
+from math import pi, sin
+from random import random
+from typing import Iterable, Tuple
+
+from palace import Buffer, Context, BaseDecoder, Device
+
+from numpy import linspace
+from numpy.random import random
+from scipy.signal import sawtooth, square, unit_impulse
+
+WAVEFORMS = {'sine': sin,
+             'square': square,
+             'sawtooth': sawtooth,
+             'triangle': lambda time: return sawtooth(time, 0.5),
+             'impulse':unit_impulse,
+             'white noise': random}
```

Comment:
`white-noise` might be a better choice to avoid user's need to escape the space.